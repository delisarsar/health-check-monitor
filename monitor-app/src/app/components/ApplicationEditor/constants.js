import React from "react";
import { gql } from "apollo-boost";
import { FormattedMessage } from "react-intl";

export const APP_NAME_CHARACTER_LIMIT = 255;
export const APP_ENDPOINT_CHARACTER_LIMIT = 255;

export const APPLICATION_NAME_LABEL = (
  <FormattedMessage
    id="features.home.edit-application.app-name"
    defaultMessage={"Application name"}
  />
);

export const APPLICATION_HEALTH_CHECK_ENDPOINT_LABEL = (
  <FormattedMessage
    id="features.home.edit-application.app-endpoint"
    defaultMessage={"Health Check Endpoint"}
  />
);

export const SAVE_LABEL = (
  <FormattedMessage
    id="features.home.edit-application.save"
    defaultMessage={"Save"}
  />
);

export const CANCEL_LABEL = (
  <FormattedMessage
    id="features.home.edit-application.save"
    defaultMessage={"Save"}
  />
);

export const DELETE_LABEL = (
  <FormattedMessage
    id="features.home.edit-application.save"
    defaultMessage={"Save"}
  />
);

export const CREATE_APPLICATION = gql`
  mutation CreateApplication($name: String!, $endpoint: String!) {
    createApplication(name: $name, endpoint: $endpoint)
  }
`;

export const DELETE_APPLICATION = gql`
  mutation DeleteApplication($applicationId: Int!) {
    deleteApplication(applicationId: $applicationId)
  }
`;

export const UPDATE_APPLICATION = gql`
  mutation UpdateApplication($name: String!, $endpoint: String!, $appId: Int!) {
    updateApplication(name: $name, endpoint: $endpoint, appId: $appId)
  }
`;

export const GET_APPLICATION_BY_ID = gql`
  query GetApplicationsById($appId: Int) {
    getApplicationById(appId: $appId) {
      id
      name
      healthCheck {
        endpoint
      }
    }
  }
`;
