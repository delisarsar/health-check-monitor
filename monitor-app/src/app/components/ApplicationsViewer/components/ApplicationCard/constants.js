export const ONLINE = "Online";
export const OFFLINE = "Offline";
export const TBD = "TBD";
