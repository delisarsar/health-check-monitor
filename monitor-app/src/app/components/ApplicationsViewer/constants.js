import React from "react";
import { FormattedMessage } from "react-intl";
import { gql } from "apollo-boost";

export const CREATE_APPLICATION_LABEL = (
  <FormattedMessage
    id="features.home.create-application"
    defaultMessage={"+ Create"}
  />
);

export const GET_APPLICATIONS_LIKE_NAME = gql`
  query GetApplicationsLikeName($name: String) {
    getApplicationsLikeName(name: $name) {
      id
      name
      healthCheck {
        endpoint
      }
    }
  }
`;
