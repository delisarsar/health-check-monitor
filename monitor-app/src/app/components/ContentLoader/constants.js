import React from "react";
import { FormattedMessage } from "react-intl";

export const GENERIC_APP_ERROR = (
  <FormattedMessage
    id="app.error"
    defaultMessage={"An error occured. Please try again"}
  />
);
