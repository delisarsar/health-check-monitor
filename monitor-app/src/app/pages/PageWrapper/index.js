import React from "react";
import PropTypes from "prop-types";
import Header from "../../components/Header";
import { Box } from "@material-ui/core";

const PageWrapper = (props) => {
  const { children } = props;
  return (
    <>
      <Header />
      <Box pt={3} pl={3}>
        {children}
      </Box>
    </>
  );
};

export default PageWrapper;

PageWrapper.propTypes = {
  children: PropTypes.node,
};
