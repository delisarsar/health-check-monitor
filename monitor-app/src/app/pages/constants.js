export const OVERVIEW_PAGE_ROUTE = "/";
export const EDIT_PAGE_ROUTE = "/edit";
export const EDIT_PAGE_APPLICATION_ID_PARAM = "applicationId";
