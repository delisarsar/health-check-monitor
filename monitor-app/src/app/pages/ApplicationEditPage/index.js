import React from "react";
import { useLocation } from "react-router-dom";
import ApplicationEditor from "../../components/ApplicationEditor";
import PageWrapper from "../PageWrapper";

const ApplicationEditPage = () => {
  const { search } = useLocation();
  return (
    <PageWrapper>
      <ApplicationEditor
        appId={new URLSearchParams(search).get("applicationId")}
      />
    </PageWrapper>
  );
};

export default ApplicationEditPage;
