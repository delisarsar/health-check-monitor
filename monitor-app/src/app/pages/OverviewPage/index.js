import React from "react";
import PageWrapper from "../PageWrapper";
import ApplicationsViewer from "../../components/ApplicationsViewer";

const OverviewPage = () => {
  return (
    <PageWrapper>
      <ApplicationsViewer />
    </PageWrapper>
  );
};

export default OverviewPage;
