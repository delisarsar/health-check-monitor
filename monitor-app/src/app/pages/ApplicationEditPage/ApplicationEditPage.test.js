import React from "react";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { IntlProvider } from "react-intl";
import messages from "../../../translations/en.json";
import { Router } from "react-router-dom";
import { EDIT_PAGE_APPLICATION_ID_PARAM } from "../../pages/constants";
import ApplicationEditPage from ".";

describe("<ApplicationEditPage />", () => {
  test("renders correctly", () => {
    const historyMock = {
      push: jest.fn(),
      location: {},
      listen: jest.fn(),
      search: { EDIT_PAGE_APPLICATION_ID_PARAM: 1 },
    };
    const { container } = render(
      <Router history={historyMock}>
        <IntlProvider locale="en-US" messages={messages}>
          <MockedProvider>
            <ApplicationEditPage />
          </MockedProvider>
        </IntlProvider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
