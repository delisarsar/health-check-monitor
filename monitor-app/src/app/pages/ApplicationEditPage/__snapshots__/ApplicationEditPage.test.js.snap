// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders the application editor with the id propely 1`] = `ShallowWrapper {}`;
